modules/ding_debt_easy/lib/Nets/Easy/Client.inc

```diff
@@ -0,0 +1,220 @@
+  }
+
+  /**
+   * Charge payment (capture) at the gateway.
+   *
+   * @param string $paymentId
+   *   The payment id to charge.
+   * @param int $amount
+   *   The amount to charge of the payment.
+   *
+   * @return string
+   *   The ID of the charge made.
+   *
+   * @throws \Nets\Easy\Exception\PaymentCommunicationException
+   * @throws \Nets\Easy\Exception\PaymentException
+   */
+  public function chargePayment($paymentId, $amount) {
+    $ch = curl_init();
+
+    // The gateway supports partial payments, but we will always pay the whole
+    // amount.
+    $payload = [
+      'amount' => $amount,
+    ];
+
+    curl_setopt_array($ch, [
+      CURLOPT_URL => $this->apiUrl . $this->apiPaymentsPath . $paymentId . '/charges',
+      CURLOPT_RETURNTRANSFER => true,
+      CURLOPT_ENCODING => "",
+      CURLOPT_MAXREDIRS => 10,
+      CURLOPT_TIMEOUT => 30,
+      CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+      CURLOPT_CUSTOMREQUEST => "POST",
+      CURLOPT_POSTFIELDS => json_encode($payload),
+      CURLOPT_HTTPHEADER => [
+        'Authorization: ' . $this->secretKey,
+        'CommercePlatformTag: DDB CMS',
+        "content-type: application/*+json",
+      ],
+    ]);
+
+    $response = curl_exec($ch);
+    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    $err = curl_error($ch);
+    curl_close($ch);
+
+    // Decode result from json payload.
+    if (201 === $code) {
+      $data = json_decode($response, TRUE);
+      return $data['chargeId'];
+    }
+
+    // Handle errors.
+    $this->genericErrorHandling($err, $code, $response);
+  }
+
+  /**
+   * Helper to handle errors.
+   *
+   * @param string $err
+   *   Error message form the http client.
+   * @param int $code
+   *   Http status code from the http client.
+   * @param string|array $response
+   *   The response from the http client.
+   *
+   * @throws \Nets\Easy\Exception\PaymentCommunicationException
+   * @throws \Nets\Easy\Exception\PaymentException
+   */
+  private function genericErrorHandling($err, $code, $response) {
+    if (!empty($err)) {
+      throw new PaymentCommunicationException($err, $code);
+    }
+
+    if (empty($response) && !in_array($code, ['400', '500'])) {
+      throw new PaymentCommunicationException('Unknown error in connection to the payment gateway (' . $code . ')', $code);
+    }
+
+    // Decode response.
+    $response = json_decode($response, TRUE);
+
+    // See https://nets-devs.isotop.se/nets-easy/en-EU/api/payment-v1/ for
+    // information about the data returned from Nets on 500 and 400.
+    if (500 === $code) {
+      if (!empty($response['message'])) {
+        throw new PaymentException($response['message'], $response['code']);
+      }
+    }
+
+    if (400 === $code) {
+      if (!empty($response['errors'])) {
+        foreach ($response['errors'] as $property => $message) {
+          throw new PaymentException($property . ' - ' . reset($message), $code);
```

Comment:
Fixed

Reply:
I do not think the optimal fix for this is to only include part of the error (the first property) in the exception. It would be preferable if the error message contained all the errors so developers do not have to fix one part only to have another part fail because of missing information in the error output.

Reply:
I have looked into the API and I can't see any where where there should be more than one error returned. Why Nets returns it in json as an array I do not known.